src/Elastic.Apm/Report/PayloadSender.cs

```diff
@@ -83,7 +83,12 @@ private async void StartWork()
 							break;
 					}
 
-					// TODO: handle unsuccesful status codes
+					if (result != null && !result.IsSuccessStatusCode)
+					{
+						var str = await result.Content.ReadAsStringAsync();
+
+						throw new Exception(str);
```

Comment:
This would end up in the catch block below...so we'd throw an exception that we catch immediately. The general strategy as an APM agent is be as non-intrusive as possible, so we should avoid throwing exceptions. I'm working on the next version of the PayloadSender in #13 , and this class doesn't really have a long future, I think a debug log would be enough here.

Reply:
Hi @gregkalapos , done